File: leetcode/2696-minimum-string-length-after-removing-substrings/2696-minimum-string-length-after-removing-substrings.py
class Solution:
    def minLength(self, s: str) -> int:
        cand = ["AB", "CD"]
        while cand[0] in s or cand[1] in s:
            s = s.replace(cand[0], "") if cand[0] in s else s.replace(cand[1], "")
        return len(s)
        

File: leetcode/2696-minimum-string-length-after-removing-substrings/2696-minimum-string-length-after-removing-substrings.java
class Solution {
    public int minLength(String s) {
        int cur = 0;
        char[] result = new char[s.length()];
        for (char c : s.toCharArray()) {
            if (cur > 0 && ((result[cur-1] == 'A' && c == 'B') || (result[cur-1] == 'C' && c == 'D'))) cur--;
            else {
                result[cur++] = c;
            }
        }
        return cur;
    }
}